### PR TITLE
[AMP] Add LLaMA 3.2 1B BFP8 Experiment

### PR DESCRIPTION
**Pull request by pglusacTT**
### Issue
N/A

### Training Workload Setting

- **Model:** LLaMA 3.2 1B
- **Strategy:** LoRA
- **Framework:** PyTorch
- **Task:** Sentiment prediction
- **Dataset:** SST2
- **Hardware:** N150

### What's Changed
This PR showcases training workload for LLaMA 3.2 1B in BFP8.

### Swiss Reports
We **reduce DRAM peak by ~16%** and have virually no accuracy drawbacks with this.
Perf remains to be tested, expecting speedups there.

#### BF16
image

#### BFP8
image

### W&B Report
image

### Checklist
- [ ] Main training workflow
- [ ] CPU testing
- [ ] GPU testing
- [ ] TT device testing
- [ ] CI testing
    - [ ] Existing tests are passing
    - [ ] Register test to CI
    - [ ] Upload golden files if necessary

### CODE REVIEW

File: blacksmith/models/torch/huggingface/hf_models.py [hunk 1]
```diff
@@ -27,6 +47,28 @@ def get_model(config: TrainingConfig, device: torch.device):
...
     model.to(device)
 
+    # Per-tensor weight dtype overrides must be registered before torch.compile
+    # so the custom_call appears in the traced graph.
+    overrides = getattr(config, "weight_dtype_overrides", None)
+    if config.use_tt and overrides:
+        from tt_torch import apply_weight_dtype_overrides
+
+        applied = apply_weight_dtype_overrides(model, overrides)
+        print(f"Applied {len(applied)} weight dtype overrides.")
```
Comment (umalesTT) [inline blacksmith/models/torch/huggingface/hf_models.py]: nit: logger instead of print

**Review by umalesTT — APPROVED** [review]
LGTM!
One question regarding memory profiling graph: What happens with const-eval? Seems like it is recomputed and deleted on each training step for bfp8 and for bf16 it is full graph?

File: blacksmith/experiments/torch/llama/xla/lora/single_chip/llama_3_2_1b_bfp8.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (mmilosevicTT) [inline blacksmith/experiments/torch/llama/xla/lora/single_chip/llama_3_2_1b_bfp8.json:1]: Why not include this in the main .yaml ? I do not see why we need both this and field `experimental_weight_dtype` in the main file

**Review by mmilosevicTT — APPROVED** [review]
LGTM, left one question

File: blacksmith/experiments/torch/llama/xla/lora/single_chip/llama_3_2_1b_bfp8.json (hunk 2 above)
Comment (pglusacTT) [reply blacksmith/experiments/torch/llama/xla/lora/single_chip/llama_3_2_1b_bfp8.json:1]: `experimental_weight_dtype` is a separate functionality to these "weight specific" overrides. I linked the docs to that, the naming is not great unfortunately.

File: blacksmith/models/torch/huggingface/hf_models.py (hunk 1 above)
Comment (pglusacTT) [reply blacksmith/models/torch/huggingface/hf_models.py]: Yes, it looks like the const-eval behaves differently when using bfp8. I'd need to explore further to give more details.
